src/content/docs/setup/configuration/cors-setup.mdx

```diff
@@ -0,0 +1,227 @@
+
+**Trailing slashes:** Origins should NOT have trailing slashes. Use `http://localhost:3000` not `http://localhost:3000/`
+
+**Protocol mismatch:** `http://localhost:3000` and `https://localhost:3000` are different origins. Match exactly.
+
+**localhost vs 127.0.0.1:** These are treated as different origins by browsers. If your storefront uses `http://127.0.0.1:3000`, add that exact URL.
+
+**Forgot to clear cache:** After changing CORS settings, always run `php bin/magento cache:flush`. Stale cache is the #1 cause of "it's not working" reports.
+
+**Using `*` with credentials:** Cannot use a wildcard `*` origin when `Allow Credentials` is enabled. Browsers will block the request.
+
+**Missing `OPTIONS` method:** Forgot to include `OPTIONS` in Allowed Methods. All CORS requests start with an OPTIONS preflight request.
+
+</Aside>
+
+## Production best practices
+
+<Options>
+
+<Option>
+
+### Same-origin architecture (recommended)
+
+The best approach for production is to avoid CORS entirely by serving both your storefront and Commerce backend from the same domain using a CDN like Fastly:
+
+- Storefront: `https://example.com` (static assets)
+- Commerce GraphQL: `https://example.com/graphql` (proxied to backend)
+
+This requires configuring your CDN to proxy `/graphql` requests to your Commerce backend.
+
+**Benefits:**
+- No CORS complexity
+- Better performance (fewer preflight requests)
+- Simpler security model
+- Better caching control
+
+**Implementation approaches:**
+
+1. **Fastly VCL routing:**
+   Configure Fastly to route GraphQL requests to your Commerce backend using VCL (Varnish Configuration Language). The VCL detects requests to `/graphql` and proxies them to your backend origin while other requests serve from your storefront origin.
+
+2. **Cloudflare Workers / CDN Edge Functions:**
+   Use edge functions to route API requests to your backend dynamically.
+
+3. **Reverse proxy (Nginx/Apache):**
+   Configure your web server to proxy `/graphql` to the backend.
+
+**Resources:**
+- <Link href="https://developer.fastly.com/reference/vcl/" text="Fastly VCL documentation" />
+- Consult your DevOps team or CDN provider for implementation
+
+**Note:** This approach requires CDN/infrastructure expertise. If you don't have these resources, use CORS configuration (Option 2) instead.
+
+</Option>
+
+<Option>
+
+### CORS configuration
+
+If you must use different domains (e.g., `storefront.example.com` → `commerce.example.com`), use the CORS configuration above with specific allowed origins (never use `*` in production).
+
+</Option>
+
+</Options>
+
+## More details
+
+For troubleshooting, configuration examples, and debug steps, see the detailed reference: [CORS Troubleshooting](/setup/configuration/cors-troubleshooting/)
+
+## References
+
+- <Link href="https://github.com/graycoreio/magento2-cors" text="graycore/magento2-cors" /> - Official module repository with configuration details
```

Comment:
Official by whom? Do we own this? I don't feel comfortable documenting this as our recommendation.